File: 1296/1296.cpp
#include <algorithm>
#include <iostream>
#include <vector>

int kadane(std::vector<int> arr) {
	int current_sum = 0; int best_sum = 0;
	for (int x : arr) {
		current_sum = std::max(x, current_sum + x);
		best_sum = std::max(current_sum, best_sum);
	}
	return best_sum;
}

int main() {
	int x, n; std::cin >> n;
	std::vector<int> arr;
	while (n--) {
		std::cin >> x;
		arr.push_back(x);
	}

	std::cout << kadane(arr) << "\n";
}

File: 1296/1296.go
package main

import (
	"fmt"
	"math"
)

func Kadane(arr []int) (best_sum int) {
	var current_sum int
	for _, x := range arr {
		current_sum = math.Max(x, current_sum+x)
		best_sum = math.Max(best_sum, current_sum)
	}
	return
}

func main() {
	var n, x int
	fmt.Scan(&n)
	arr := make([]int, n)
	for ; n > 0; n-- {
		fmt.Scan(&x)
		arr = append(arr, x)
	}
	fmt.Println(Kadane(arr))
}
